Add initial plugin scaffolding

econchick opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines <https://github.com/spotify/gordon-gcp/blob/master/CONTRIBUTING.rst>
2. If the PR is unfinished, please prefix the subject line with [WIP], [DRAFT], or [RFC].
-->

**What this PR does / why we need it**:

Sets up how the plugin paths need to be when installed with [gordon](https://github.com/spotify/gordon). 

This also adds a dev dependency to auto-reload when docs are edited.

All the names and the repo organization in general is meant to be a placeholder for what we actually decide on in terms of naming convention, interface structure, etc.

**Which issue(s) this PR fixes** 
<!-- optional; in `fixes #<issue number>, fixes #<issue_number>, ...` format, will close the issue(s) when PR gets merged: -->

N/A

**Special notes for your reviewer**:

This will fail because it makes an assumption that interfaces are defined in gordon core when they're not yet.

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, start the release note with the string "action required: ".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

cc @spotify/alf

econchick commented on docs/providers.rst:
```diff
@@ -0,0 +1,111 @@
+--------------------------------------------
+
+.. literalinclude:: ../gordon.toml.example
+   :language: ini
```
The `literalinclude` sphinx directive includes the file itself and highlights it according to `language` defined (from pygments).

econchick commented on docs/providers.rst:
```diff
@@ -0,0 +1,111 @@
+Plugin Configuration
+--------------------
+
+.. _gcp:
```
These are custom references. Instead of saying `:ref:[gcp]` (and maybe needing to escape some backticks) I can just say `:ref:gcp` when I want to link here (which is done below).

smstone commented on docs/providers.rst:
```diff
@@ -0,0 +1,111 @@
+
+    Configuration defined for the ``compute`` provider will overwrite values of the same keys defined under :ref:`gcp <gcp>`, then inherit the rest.
+
+    The ``compute`` provider does **not** have access to configuration for :ref:`gcp.dns <gcpdns>` and :ref:`gcp.pubsub <gcppubsub>`.
```
just a suggestion but might want to just add a note at the top saying this, rather than saying it in every description?

econchick replied:
ah yeah good call.

econchick commented:
@smstone thanks for the review - I just moved up the `note` to the beginning of the plugin config and removed the individual ones. I also changed it from a `note` to `attention` so it's more obvious (red background instead of gray):

![2018-01-02 at 9 30 am](https://user-images.githubusercontent.com/1191069/34486823-90541bb4-ef9f-11e7-8fbf-5338041c9eae.png)

I'm going to merge this since you gave the +1, but let me know if you think anything else needs to be changed.
